### ISSUE TITLE
Ability to use route function and route string together

### ISSUE THREAD

**Issue by leonmondria** [vikejs/vike#370]
### Description

As a developer using vite-plugin-ssr I want to re-use routing internals so that I don't have to roll my own route matching.

I have some cases where I want more control over the routing, I use the Route Function for that. 
Sometimes I still want to use some form of string matching within the function, I use regex for that.
It would be nice to also be able to re-use the built-in Route String functionality, something like this:

```import { ServerPageContext } from '~/renderer/types';
import { resolveRouteString } from 'vite-plugin-ssr';

const ROUTE = '/product/@id/@slug';

export default (pageContext: ServerPageContext) => {
  const match = resolveRouteString(ROUTE, pageContext.url);
  if(!match) return false;

  const { routeParams } = match;

  // do some extra stuff

  return {
    routeParams,
    precedence: 1,
  };
};
```

I'm running on version 0.4.9.
Would be nice to hear your thoughts on this!

**Comment by leonmondria** [comment]
In my case I want to enable/disable pages feature-based:

```
if (!features.vue3?.giftcard) {
    return false;
  }
```

The features are retrieved from API in onBeforeRender of _default.page.server and passed to the pageContext.

**Comment by brillout** [comment]
Makes sense. I'm up for implementing this.

I'm thinking a `resolveRouteString()` that returns a value that aligns well with Route Functions:

```js
export default '/product/@id'
// Equivalent to the above:
import { resolveRouteString } from 'vite-plugin-ssr/routing'
export default pageContext => resolveRouteString('/product/@id', pageContext.url)
```

```js
export default (pageContext) => {
  const res = resolveRouteString('/product/@id', pageContext.url)
  if (res.match) {
    // do some extra stuff
  }
  return res
};
```

For features that represent a unique value proposition (compared to Next.js/Nuxt) I usually ask for a small monetary compensation. Do you mind me asking whether you develop your app for yourself or a company?

Feel free to DM me on Discord.

**Comment by brillout** [comment]
Released in `0.4.21`.

Docs: https://vite-plugin-ssr.com/route-function#resolveroute.
